code/modules/atmospherics/machinery/components/fusion/hfr_procs.dm

```diff
@@ -159,7 +159,20 @@
  * Getter for fusion fuel moles
  */
 /obj/machinery/atmospherics/components/unary/hypertorus/core/proc/check_fuel()
-	return (internal_fusion.gases[/datum/gas/tritium][MOLES] > FUSION_MOLE_THRESHOLD && internal_fusion.gases[/datum/gas/hydrogen][MOLES] > FUSION_MOLE_THRESHOLD)
+	if(!selected_fuel)
+		return FALSE
+	if(!internal_fusion.total_moles())
+		return FALSE
+	var/gas_check = 0
+	for(var/gas_type in selected_fuel.requirements)
+		internal_fusion.assert_gas(gas_type)
+		if(gas_type == selected_fuel.requirements[3])
+			continue
+		if(internal_fusion.gases[gas_type][MOLES] >= FUSION_MOLE_THRESHOLD)
+			gas_check++
+	if(gas_check == selected_fuel.requirements.len - 1)
```

Comment:
????????????????

Reply:
Same as above there are three entries but only two are actual fuel (might change of i switch to another datum)